src/function-builder.ts

```diff
@@ -295,6 +295,10 @@ export function runWith(runtimeOptions: RuntimeOptions): FunctionBuilder {
     return new FunctionBuilder(runtimeOptions);
   }
 }
+  
+type ExtractParams<T> = T extends `${infer Pre}{${infer A}}${infer Post}`
+  ? { [K in A]: string } & ExtractParams<Pre> & ExtractParams<Post>
+  : string extends T ? Record<string,string> :  {};
```

Comment:
```suggestion
  :  {};
```
Update to this to force the `context.params` to be `{}` if no params found - aka more strict